### PR TITLE
feat(exit): non-zero exit code for getUnusedRules

### LINKED ISSUE #31: Exit status
Currently exit status is 0, the original module used process.exit(1), I removed that on purpose, I did not want the error to scroll past the viewport.

Should there be an option `--exit=1` default to `0`?
May be in [CI environment](https://github.com/watson/is-ci) `exit` status is 1?

Thoughts?

### PR DESCRIPTION
**Pull request by sarbbottam**
use -n|no-error to suppress
fixed #31 

/ @ta2edchimp

### CODE REVIEW

File: test/bin.js [hunk 1]
```diff
@@ -28,11 +29,13 @@ describe('bin', function() {
...
     }
+    process.exit = function noop() {}
     process.argv = process.argv.slice(0, 2)
   })
 
   afterEach(function() {
     console.log = consoleLog // eslint-disable-line no-console
+    process.exit = processExit
     // purge yargs cache
     delete require.cache[require.resolve('yargs')]
```
Comment (ta2edchimp) [inline test/bin.js]: Shouldn't `proxyquire` take care of the cache?

From its docs:

```JavaScript
// ensure we don't get any module from the cache, but to load it fresh every time 
var proxyquire = require('proxyquire').noPreserveCache();
```

Comment (ta2edchimp) [inline test/bin.js]: Respectively, extend the stub with

```JavaScript
  'yargs': {
    '@global': true
  }
```

?

At the end of the day, if it works (and it does), I'm fine with it. Just asked for clarification ...

Comment (sarbbottam) [inline test/bin.js]: Are you suggesting something like below?

```diff
var assert = require('assert')
-var proxyquire = require('proxyquire')
+var proxyquire = require('proxyquire').noPreserveCache();
 var sinon = require('sinon')

 var consoleLog = console.log // eslint-disable-line no-console
@@ -19,6 +19,9 @@ var stub = {
       getUnusedRules: getUnusedRules,
     }
   },
+  'yargs': {
+    '@global': true
+  }
 }

 describe('bin', function() {
@@ -36,8 +39,6 @@ describe('bin', function() {
   afterEach(function() {
     console.log = consoleLog // eslint-disable-line no-console
     process.exit = processExit
-    // purge yargs cache
-    delete require.cache[require.resolve('yargs')]
   })

   it('option -c|--current', function() {
```

could not get it working, looks like `yargs` is cached after the first `require`. Let me know if I have mis interpreted.

Comment (ta2edchimp) [inline test/bin.js]: Yep, that is exactly what I thought of. I just remembered, that I had an issue with the caching behavior in another case ... Let's forget this. The explicit deletion from `require.cache` seems to work, so let's not overcomplicate this.

_(I have to dig into this deeper some time, there has to be a way to accomplish this with proxyquire's own means... if not, it has to be a bug.)_

**Comment by sarbbottam** [comment]
Is it good to merge then?

**Comment by ta2edchimp** [comment]
LGTM! :+1:

**Pull request merged**